rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/util/Lazy.java

```diff
@@ -28,9 +28,10 @@
 
 import java.util.function.Supplier;
 import org.checkerframework.checker.nullness.qual.MonotonicNonNull;
+import org.checkerframework.checker.nullness.qual.NonNull;
 
-public interface Lazy<T> extends Supplier<T> {
-    public static <T> Lazy<T> defer(Supplier<T> generator) {
+public interface Lazy<T extends @NonNull Object> extends Supplier<T> {
```

Comment:
Interesting, I did not know you could (ab)use type params to check for non-null objects!

Reply:
well, this is a thing, you cannot put annotations on type parameters, so if you want to say, hey the type here will not return a null value, than this is the way to mark it, but in a way, this is more of a CF thing than something else, and I'm unsure how much we want to spread this around.